scripts/setup.js

```diff
@@ -0,0 +1,178 @@
+  writeJSON(packageJSON, 'package.json')
+}
+
+// replace default plugin name and url with project name and dist file
+function updateJBrowseConfig(tsdxName, pluginName) {
+  const jbrowseConfig = require('../jbrowse_config.json')
+  jbrowseConfig.plugins[0].name = pluginName
+  jbrowseConfig.plugins[0].url = `http://localhost:9000/dist/${tsdxName}.umd.development.js`
+  writeJSON(jbrowseConfig, 'jbrowse_config.json')
+}
+
+// replace default plugin name and url with project name and dist file
+function updateExampleFixture(tsdxName, pluginName) {
+  const exampleFixture = require('../cypress/fixtures/hello_view.json')
+  exampleFixture.plugins[0].name = pluginName
+  exampleFixture.plugins[0].url = `http://localhost:9000/dist/${tsdxName}.umd.development.js`
+  writeJSON(exampleFixture, 'cypress/fixtures/hello_view.json')
+}
+
+// create a dot directory for the jbrowse build to live in
+function makeJBrowseDir() {
+  if (!fs.existsSync('.jbrowse')) {
+    fs.mkdirSync('.jbrowse')
+  }
+}
+
+function setupGithubAction(packageJSON, projectName) {
+  // move integration test into workflow folder
+  if (!fs.existsSync(path.join('.github', 'workflows'))) {
+    fs.mkdirSync(path.join('.github', 'workflows'), { recursive: true })
+  }
+  fs.renameSync(
+    'integration.yml',
+    path.join('.github', 'workflows', 'integration.yml'),
+  )
+
+  // add status badge to README
+  const repoUrl = parseRepoUrl(packageJSON.repository)
+  if (repoUrl !== undefined) {
+    let README = readFile('README.md').split(/\r?\n/)
+    README[0] = `# ${projectName} ![Integration](${repoUrl}/workflows/Integration/badge.svg?branch=main)`
+    fs.writeFileSync('README.md', README.join(os.EOL), 'utf8')
+  }
+}
+
+/* 
+****************************
+Helpers
+****************************
+*/
+
+function writeJSON(data, path) {
+  try {
+    fs.writeFileSync(path, `${JSON.stringify(data, null, 2)}\n`)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+function readFile(path) {
+  try {
+    return fs.readFileSync(path, 'utf8')
+  } catch (err) {
+    console.error(err)
+    return false
+  }
+}
+
+// snagged from https://stackoverflow.com/a/53952925
+function toPascalCase(string) {
+  return `${string}`
+    .replace(new RegExp(/[-_]+/, 'g'), ' ')
+    .replace(new RegExp(/[^\w\s]/, 'g'), '')
+    .replace(
+      new RegExp(/\s+(.)(\w+)/, 'g'),
+      ($1, $2, $3) => `${$2.toUpperCase() + $3.toLowerCase()}`,
+    )
+    .replace(new RegExp(/\s/, 'g'), '')
+    .replace(new RegExp(/\w/), s => s.toUpperCase())
+}
+
+function getTsdxPackageName(projectName) {
+  // From TSDX utils
+  return projectName
+    .toLowerCase()
+    .replace(/(^@.*\/)|((^[^a-zA-Z]+)|[^\w.-])|([^a-zA-Z0-9]+$)/g, '')
+}
+
+function parseRepoUrl(repo) {
+  let url
+  if (repo !== undefined) {
+    if (typeof repo === 'string') {
+      url = repo
+    } else if (typeof repo === 'object') {
+      url = repo.url
+    }
+  }
+
+  if (typeof url === 'string' && url.includes('github.com')) {
```

Comment:
The includes check is done on the next line now.

```suggestion
  if (typeof url === 'string')) {
```